### DISCUSSION TITLE
Remix integration

### DISCUSSION THREAD

**Discussion by moishinetzer** [tauri-apps/tauri#5987] [Ideas]
Since Remix.run has similar functionalities to Next.js, and it could be compared to a SvelteKit for React, it seems like it would be a perfect fit for Turi.

Is there any interest in creating this integration from the team?

**Comment by FabianLars** [comment]
Currently not at all. As long as Remix requires a server/nodejs it's just not a good fit for tauri since there is no nodejs runtime in a tauri app.

**Reply by treefitty** [reply]
Pardon my potential ignorance, but as Remix packages to both server and client, can't you package the client with Tauri? ... or have I missed something really obvious? 😅

**Reply by FabianLars** [reply]
Remix was clearly designed to be used with a server, and trying to use it without it is _in my opinion_ not worth the headaches. Also, as i understand it, its "default" output expects a server so just taking the client part and stuffing it into Tauri won't help much as you still need a server which would then serve the files again (ignoring the local files). This sounds like a feature relatively easy to add in Remix but again it's not really in scope for it so i doubt they'd invest time into it. -> Could be wrong about that stuff, i'm not an active Remix user.

Anywayyy, after my comment above i learned that you _can_ get actual SSG output out of Remix: https://twitter.com/mjackson/status/1585795441907494912?s=20. But with the mentioned drawbacks and additional build steps i stand with my "it's just not a good fit for tauri" statement. at least for now.

**Reply by treefitty** [reply]
Hang on, the Remix client is just bundled of HTML, CSS and JS.
The server is bundled and gets deployed to the cloud (however you want, AWS, Vercel etc). 
The client communicates to the server via API ala Figma, Miro etc etc

I'm still not seeing why this is impossible given the blurb for Tauri says "Developers can integrate any front-end framework that compiles to HTML, JS and CSS for building their user interface" as that's all Remix is doing

**Reply by wildeyes** [reply]
@paulcarroll, remix calls itself a full stack web framework, not an FE framework.
Definitely out of scope for Tauri's main to integrate Remix run @paulcarroll @moishinetzer, but perhaps if you're interested in creating a community solution I'd love to help.

**Reply by FabianLars** [reply]
Your comment made me notice that i never sent my response... I can't exactly remember what i drafted up but the gist was something like this:
> Hang on, the Remix client is just bundled of HTML, CSS and JS.
> ... HTML, JS and CSS for building their user interface" as that's all Remix is doing

That's not quite the case. If you run `remix build` it will output javascript code, yes, but it's not javascript Tauri or any browser can process, also note that there won't be any explicit html file at this point. This code is specifically meant to be consumed by Remix' server runtime which uses this code to build the client side HTML/CSS/JS. It does not matter if any of your routes are pre-rendered since the prerendered stuff is still "inside" that client-side code and Tauri (or any browser) can't just extract that.

Again, i'm not saying that this can't ever be possible (also note the twitter link i sent), it's just not something that works with Remix _current_ output. Just try to build your Remix app, look at the build output files, and/or try to serve it _without_ the Remix server to see what i'm trying to say.

Or am i just missing your point completely?

**Reply by treefitty** [reply]
Ahh right, yes that makes sense. When I wrote my initial comments I'd only barely glanced at Remix, but was ticking off some items in my head of what I wanted to achieve with my project before I went too far down the rabbit hole. 
What you say indeed makes a 100% sense now.
As it turns out you can use client side rendering, but that rapidly descends into an abyss of madness from what I can see. 

Appreciate the follow up comments @FabianLars and @wildeyes

**Reply by gitaugakwa** [reply]
The team is currently working on adding a client-side layer that would sit on top of the server-side loaders and actions.

https://github.com/remix-run/remix/discussions/7634

**Comment by omakise** [comment]


**Reply by FabianLars** [reply]
I don't understand your `+1` request here to be honest. What do you want us to do (taking into account the rest of the discussion above)?

It sounds like you want to host the remix app on a remote server, which isn't really what this discussion is about, _but_ this should be possible without much of a problem.

**Comment by m-abdelwahab** [comment]
Remix just add support for SPA mode https://remix.run/docs/ja/main/future/spa-mode

Would be cool to see how it can work with Tauri 👀

**Reply by FabianLars** [reply]
Step one, create the remix project as described in your link.
Step two, follow https://tauri.app/v1/guides/getting-started/setup/integrate. In my case the answers to the tauri init questions looked like this:
```
✔ What is your app name? · my-remix-app
✔ What should the window title be? · my-remix-app
✔ Where are your web assets (HTML/CSS/JS) located, relative to the "<current dir>/src-tauri/tauri.conf.json" file that will be created? · ../build/client
✔ What is the url of your dev server? · http://localhost:5173
✔ What is your frontend dev command? · pnpm remix vite:dev
✔ What is your frontend build command? · pnpm remix vite:build
```
Step 3, run `tauri dev` and/or `tauri build`. These commands should now work as expected and will automatically build the frontend project (or run it in dev mode).

**Reply by raymclee** [reply]
this works, however, I need to close the app and restart it to see the new route. otherwise, it will show 404
